File: CSharpPractice/LeetcodePractice/Leetcode1593.cs
public class Solution1593 {
    public HashSet<string> theSet = new HashSet<string>();
    public string theS;
    public int n;

    public int helper(int start)
    {
        if(start==n){return theSet.Count;}
        int ans=0;
        for(int end=start+1;end<n+1;end++)
        {
            string curString =theS.Substring(start,end-start);
            if(!theSet.Contains(curString))
            {
                theSet.Add(curString);
                ans=Math.Max(ans,helper(end));
                theSet.Remove(curString);
            }
        }
        return ans;
    }
    public int MaxUniqueSplit(string s) 
    {
        theS=s;
        n=s.Length;
        return helper(0);
    }
}

File: JavaScriptPractice/Leetcode1593.js
let theSet;
let n;
let theS;
var helper = function(start)
{
    if(start==n){return theSet.size;}
    let ans=0;
    for(let end=start+1;end<n+1;end++)
    {
        let curString = theS.slice(start,end);
        if(!theSet.has(curString))
        {
            theSet.add(curString);
            ans=Math.max(ans,helper(end));
            theSet.delete(curString);
        }
    }
    return ans;
}
var maxUniqueSplit = function(s) 
{
    theS=s;
    n=s.length;
    theSet = new Set();
    return helper(0);  
};